### ISSUE TITLE
HTTP::Server specs fail on OpenSSL 1.1

### ISSUE THREAD

**Issue by mniak** [crystal-lang/crystal#6979]
### Steps to reproduce
```
make libcrystal
bin/crystal spec/std/http/server/server_spec.cr
```
### Expected
Success

### But got
```
................................F.....

Failures:

  1) HTTP::Server handles exception during SSL handshake (#6577)
     Failure/Error: expect_raises(OpenSSL::SSL::Error) do

       Expected OpenSSL::SSL::Error, got #<Errno: SSL_connect: Success> with backtrace:
         # src/openssl.cr:115:15 in 'initialize'
         # src/openssl.cr:100:7 in 'new'
         # src/openssl/ssl/socket.cr:34:9 in 'initialize'
         # src/openssl/ssl/socket.cr:3:5 in 'new'
         # spec/std/http/server/server_spec.cr:517:13 in '->'
         # src/spec/methods.cr:255:3 in 'it'
         # spec/std/http/server/server_spec.cr:494:5 in '->'
         # src/spec/context.cr:255:3 in 'describe'
         # src/spec/methods.cr:16:5 in 'describe'
         # spec/std/http/server/server_spec.cr:234:3 in '__crystal_main'
         # src/crystal/main.cr:97:5 in 'main_user_code'
         # src/crystal/main.cr:86:7 in 'main'
         # src/crystal/main.cr:106:3 in 'main'
         # __libc_start_main
         # _start
         # ???

     # spec/std/http/server/server_spec.cr:516

Finished in 422.48 milliseconds
38 examples, 1 failures, 0 errors, 0 pending

Failed examples:

crystal spec spec/std/http/server/server_spec.cr:494 # HTTP::Server handles exception during SSL handshake (#6577)
```

### git branch: `master`
commit `fc52166cad58f224e9d54a2d83195702401db75c` (currently the last on) 

#### `crystal -v`
```
Crystal 0.26.1 (2018-09-27)

LLVM: 6.0.1
Default target: x86_64-pc-linux-gnu
```
#### `bin/crystal -v`
```
Crystal 0.26.1 (2018-09-27)

LLVM: 6.0.1
Default target: x86_64-pc-linux-gnu
```
#### `openssl version`
`OpenSSL 1.1.1  11 Sep 2018`

#### `uname -a`
```
Linux MyHostname 4.18.9-arch1-1-ARCH #1 SMP PREEMPT Wed Sep 19 21:19:17 UTC 2018 x86_64 GNU/Linux
```

## Additional notes
The error happened originally by running `make std_spec`. I'm running the *spec* directly and the error is the same.

**Comment by RX14** [comment]
I'm interested - can anyone print the error returned by `expect_raises(OpenSSL::SSL::Error)` on a platform where the spec passes?

**Comment by Sija** [comment]
Fixed by #7068?

**Comment by Sija** [comment]
@RX14 Shouldn't this be closed?
